lib/src/modal/modal.component.ts

```diff
@@ -0,0 +1,102 @@
+import { Component, Input, ViewEncapsulation, ComponentRef, HostListener, ElementRef, ViewChild } from "@angular/core";
+import { SebModalBackdropComponent } from "./modal.backdrop";
+import { ModalService } from "./modal.service";
+import { ModalSizeType, ModalPositionType } from "./modal.type";
+import { AnimationEvent } from "@angular/animations";
+import { fadeInAnimation } from "./animation";
+
+@Component({
+    selector: "sebng-modal",
+    styleUrls: ["./modal.component.scss"],
+    templateUrl: "./modal.component.html",
+    encapsulation: ViewEncapsulation.None,
+    animations: [fadeInAnimation],
+})
+export class ModalComponent {
+    @Input() id?: string;
+    @Input() size?: ModalSizeType;
+    @Input() center?: boolean;
+    @Input() position: ModalPositionType;
+    @Input() fullscreen?: boolean;
+    @Input() backdropDismiss?: boolean = true;
+    @Input() escapeKeyDismiss?: boolean = true;
+    @Input() className?: string;
+    @Input() ariaLabel?: string;
+    @Input() ariaDescribedby?: string;
+    @Input() animationDuration?: string = ".15s";
+    @ViewChild("modalRef") modalRef: ElementRef;
+    backDropRef: ComponentRef<SebModalBackdropComponent>;
+    toggle: boolean; // toggle is required to enable the open or close animation
+
+    constructor(private modalService: ModalService) {}
+
+    /**
+     * construct the class names that needs to be appended to the modal depending on the inputs requested
+     * @returns { [key: string]: boolean } key value pair for the positions of the modal
+     */
+    get modalPosition(): { [key: string]: boolean } {
+        return {
+            "modal-aside": !!this.position && !this.fullscreen,
+            "modal-aside-left": this.position === "left",
+            "modal-aside-right": this.position === "right",
+            "modal-fullscreen": this.fullscreen,
+        };
+    }
+
+    /**
+     * emit close event when backdrop is clicked
+     */
+    @HostListener("click", ["$event"])
+    onBackdropClick(event: MouseEvent): void {
+        if (this.backdropDismiss) {
+            if (event && event.target && event.target["classList"] && event.target["classList"].length) {
+                const classList: DOMTokenList = event.target["classList"];
+                if (classList.contains("modal")) {
+                    this.close();
+                }
+            }
+        }
+    }
+
+    /**
+     * emit close when escape key is clicked
+     */
+    @HostListener("keyup", ["$event"])
+    onEscKey(event: KeyboardEvent): void {
+        if (this.escapeKeyDismiss && event.keyCode === 27) {
```

Comment:
keyCode is deprecated. https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode#:~:text=The%20deprecated%20KeyboardEvent.,value%20of%20the%20pressed%20key.&text=You%20should%20avoid%20using%20this,Instead%2C%20you%20should%20use%20KeyboardEvent.

Reply:
use .key instead

Reply:
done